tizen-api/src/tizen-api-pipeline.c

```diff
@@ -331,9 +420,23 @@ nns_pipeline_start (nns_pipeline_h pipe)
 int
 nns_pipeline_stop (nns_pipeline_h pipe)
 {
-  /** @todo NYI */
+  nns_pipeline *p = pipe;
+  GstStateChangeReturn scret;
+  GstState state, pending;
 
-  return NNS_ERROR_NONE;
+  g_mutex_lock (&p->lock);
+  scret = gst_element_set_state (p->element, GST_STATE_PAUSED);
+  g_mutex_unlock (&p->lock);
+
+  if (scret == GST_STATE_CHANGE_SUCCESS || scret == GST_STATE_CHANGE_ASYNC)
```

Comment:
check against GST_STATE_CHANGE_NO_PREROLL as well? or simply scret != GST_STATE_CHANGE_FAILURE

Reply:
Ok. I've decided not to be a paranoid and believe that ```gst_element_set_state(p->element, GST_STATE_PAUSED)``` will be ok as long as it has not returned FAILURE. :)